Use cygpath if present to create top level dir for Cygwin/MSYS2

pnetherwood opened the pull request:
git rev-parse --show-topdir command on Git for Windows returns a Windows path which cannot be found if using Vim and Vimagit from Cygwin or MSYS2. This change checks for the presence of the cygpath command if present is used to convert the windows path into a Unix style path compatible with Cygwin and MSYS2.

Commit:
Use cygpath if present to create top level dir for Cygwin/MSYS2 compatibility

jreybert commented:
Please use tabs instead of spaces for indentation.

Also, your commit message is weird. Put the whole title on first line, and add your PR comment in commit body.

pnetherwood commented:
Ok. Will do.

On Fri, 3 Jan 2020 at 12:50, Jérôme Reybert <notifications@github.com>
wrote:

> Please use tabs instead of spaces for indentation.
>
> Also, your commit message is weird. Put the whole title on first line, and
> add your PR comment in commit body.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/jreybert/vimagit/pull/183?email_source=notifications&email_token=AC2VNB2PN7UNBUC6LIXCG3TQ34YCVA5CNFSM4KCMJB6KYY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOEIBBZ6A#issuecomment-570563832>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AC2VNBZNF7ZWZK4YHXAF6FLQ34YCVANCNFSM4KCMJB6A>
> .
>

Commit:
Make top level git path work with Cygwin/MSYS2

Cygwin and MSYS2 used Git for Windows which returns a windows path which
doesn't work in vim.
This change looks for the presence of the cygpath executable which
if present indicates its running under Cygwin or MSYS2.
  It then uses cypath is to convert the windows path to a compatible
  path.
